lib/src/widgets/adw/raw_target.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:flutter/material.dart';
+
+/// Set of status that a [RawAdwTarget] widget can be at any given time.
+enum TargetStatus { enabled, active, hover, tapDown }
+
+typedef TargetColorBuilder = Color? Function(BuildContext, TargetStatus);
+
+typedef TargetWidgetBuilder = Widget Function(BuildContext, TargetStatus);
+
+/// [RawAdwTarget] is a widget used as a base to build Adwaita-style widgets
+/// with pressed and hover properties.
+///
+/// Widgets that use [RawAdwTarget] as their base can rebuild
+/// their background through the [backgroundColorBuilder] callback.
+class RawAdwTarget extends StatefulWidget {
```

Comment:
I'm afraid the comment I'm about to make is a slight case of [the bike shed conundrum](https://bikeshed.com/), so feel free to disregard it and keep in mind that I think this commit is already very good! I won't be offended :).

I'm not familiar with any library naming interfaces/base widget `Raw`. This naming created a slight confusion for me when I read the commit, and it took me a couple of minutes to understand that it was an interface and not a type of widget I should use. I've tried to find an equivalent in libadwaita but couldn't find anything, and I think the closest is really [`GtkButton`](https://docs.gtk.org/gtk4/class.Button.html). I'm proposing to name such Widget `AdwButton`. Because in the end, that is what the class is, isn't? Other buttons that you plan to implement will all inherit from this class, right?

What do you think? If it's too much work for now, or you disagree, feel free to disregard this comment, but I think it would help with reading the code :).

Reply:
Thank you very much for the comment. I also think that `RawAdwTarget` could be a bit confusing.

After reading your comment, `AdwButton` looks like a good name to me as well. All 'clickable' elements are buttons in a way, so from that side makes sense.

I'll update the name shortly. And thanks for all the comments, this way we're improving the package immensely :)

Reply:
Sounds good to me :)